integration/solana/errors.spec.ts

```diff
@@ -1,42 +1,34 @@
 // SPDX-License-Identifier: Apache-2.0
 
-import { TransactionError } from '@solana/solidity';
+import { BN } from '@project-serum/anchor';
 import expect from 'expect';
 import { loadContract } from './setup';
 
 describe('Testing errors', function () {
     this.timeout(500000);
 
     it('errors', async function () {
-        const { contract } = await loadContract('errors');
+        const { program, storage } = await loadContract('errors');
 
-        let res = await contract.functions.do_revert(false);
+        let res = await program.methods.doRevert(false).view();
 
-        expect(Number(res.result)).toEqual(3124445);
+        expect(res).toEqual(new BN(3124445));
 
         try {
-            res = await contract.functions.do_revert(true, { simulate: true });
-        } catch (e) {
-            expect(e).toBeInstanceOf(TransactionError);
-            if (e instanceof TransactionError) {
-                expect(e.message).toBe('Do the revert thing');
-                expect(e.computeUnitsUsed).toBeGreaterThan(1400);
-                expect(e.computeUnitsUsed).toBeLessThan(1600);
-                expect(e.logs.length).toBeGreaterThan(1);
-            }
+            res = await program.methods.doRevert(true).simulate();
+        } catch (e: any) {
+            const logs = e.simulationResponse.logs;
+            expect(logs).toContain('Program log: Going to revert');
```

Comment:
Because anchor is not decoding the return value when the contract does not return success.

In fact, anchor idl have no method for specifying the encoding of errors.

Reply:
> Because anchor is not decoding the return value when the contract does not return success.
> 
> In fact, anchor idl have no method for specifying the encoding of errors.

Is this something we would like Anchor to have in the future?
Perhaps, it deservers a comment on the docs.

Reply:
I've added a comment in the documentation for this and created https://github.com/hyperledger/solang/issues/1152

I don't think this is likely to be changed in anchor. The revert reason should really go into the log, and not be encoded in the return data.